library/include/modules/Gui.h

```diff
@@ -125,12 +127,24 @@ namespace DFHack
         // Show an announcement with effects determined by announcements.txt
         DFHACK_EXPORT void showAutoAnnouncement(df::announcement_type type, df::coord pos, std::string message, int color = 7, bool bright = true, df::unit *unit1 = NULL, df::unit *unit2 = NULL);
 
+        // Process an announcement exactly like DF would, which might result in no announcement
+        DFHACK_EXPORT bool autoDFAnnouncement(df::report_init r, std::string message);
+        DFHACK_EXPORT bool autoDFAnnouncement(df::announcement_type type, df::coord pos, std::string message, int color = 7, bool bright = true,
+                                              df::unit *unit1 = NULL, df::unit *unit2 = NULL, bool is_sparring = false);
+
         /*
          * Cursor and window coords
          */
         DFHACK_EXPORT df::coord getViewportPos();
         DFHACK_EXPORT df::coord getCursorPos();
 
+        // Recenter the viewscreen, based on DF code for announcements and scrolling
+        DFHACK_EXPORT void pauseRecenter(int32_t x, int32_t y, int32_t z, bool pause);
+        DFHACK_EXPORT inline void pauseRecenter(df::coord pos, bool pause) { return pauseRecenter(pos.x, pos.y, pos.z, pause); }
```

Comment:
why is there a `return` here? `pauseRecenter(x, y, z, pause)` is a `void` function